content-services/content-sources-backend
# Fixes 4742: snapshot and bulk delete endpoints and task

## Summary
- Two endpoints for manual snapshot deletion added, one for deleting a single snapshot and a bulk delete one.
- Endpoints only soft-delete snapshots and then start a task cleaning those up fully along with Pulp content and updating templates that are using those.

## Testing steps
1. Create a repository, with snapshots enabled.
2. Add the repository to a template and make sure it uses the snapshots that's gonna be deleted.
3. After the repository has a snapshot created, change the URL and trigger a new snapshot.
4. Delete a snapshot from that repo with the new endpoint.
`http DELETE :8000/api/content-sources/v1.0/repositories/${REPO_UUID}/snapshots/${SNAP_UUID}`
5. Verify that the snapshot has been deleted (i.e. not present when listing snapshots for a repo).
6. Verify that the template snapshot has been updated to the other one. (Doesn't work yet.)
- For bulk delete, do the same with more snapshots.
